File: src/controllers/symptomController.js
const Symptom = require('../models/Symptom');

const getSymptoms = async (req, res, next) => {
  try {
    const symptoms = await Symptom.find({ userId: req.user._id }).sort({ date: -1 });
    res.json({ success: true, symptoms });
  } catch (err) { next(err); }
};

const createSymptom = async (req, res, next) => {
  try {
    const { name, severity, date, notes } = req.body;
    if (!name) return res.status(400).json({ success: false, message: 'Tên triệu chứng là bắt buộc' });
    const symptom = await Symptom.create({ userId: req.user._id, name, severity, date, notes });
    res.status(201).json({ success: true, symptom });
  } catch (err) { next(err); }
};

const deleteSymptom = async (req, res, next) => {
  try {
    const symptom = await Symptom.findOneAndDelete({ _id: req.params.id, userId: req.user._id });
    if (!symptom) return res.status(404).json({ success: false, message: 'Không tìm thấy triệu chứng' });
    res.json({ success: true, message: 'Đã xóa triệu chứng' });
  } catch (err) { next(err); }
};

module.exports = { getSymptoms, createSymptom, deleteSymptom };

File: src/main/java/com/hi/api/controller/SymptomController.java
package com.hi.api.controller;

import com.hi.api.dto.request.CreateSymptomRequest;
import com.hi.api.model.Symptom;
import com.hi.api.model.User;
import com.hi.api.service.SymptomService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/symptoms")
public class SymptomController {

    private final SymptomService symptomService;

    public SymptomController(SymptomService symptomService) {
        this.symptomService = symptomService;
    }

    @GetMapping
    public ResponseEntity<Map<String, Object>> getSymptoms(@AuthenticationPrincipal User user) {
        List<Symptom> symptoms = symptomService.getSymptoms(user.getId());
        return ResponseEntity.ok(Map.of("success", true, "symptoms", symptoms));
    }

    @PostMapping
    public ResponseEntity<Map<String, Object>> createSymptom(
            @AuthenticationPrincipal User user,
            @Valid @RequestBody CreateSymptomRequest req) {
        Symptom symptom = symptomService.createSymptom(user.getId(), req);
        return ResponseEntity.status(HttpStatus.CREATED).body(Map.of("success", true, "symptom", symptom));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, Object>> deleteSymptom(
            @AuthenticationPrincipal User user,
            @PathVariable String id) {
        symptomService.deleteSymptom(user.getId(), id);
        return ResponseEntity.ok(Map.of("success", true, "message", "Đã xóa triệu chứng"));
    }
}
